### DISCUSSION TITLE
EnumTypeName(i) (mapping integer -> Enumeration value is available?

### DISCUSSION THREAD

**Discussion by mpradovelasco** [OpenModelica/OpenModelica#10679] [Q&A] [answered]
Hello everyone!

Is the mapping function "integer to enumeration value" available in OM 1.21? I refer to the conversion function EnumTypeName(i) described in the modelica specification (Section 3.7.1 in modelica version 3.4). 

Thank you again for all your help.

**Comment by perost** [comment]
No, we have gotten around to implementing that yet, I think you might be the first to ask about it. But I can take a look at it next week maybe, it shouldn't be a lot of work to implement.

**Reply by mpradovelasco** [reply]
It is curious that I could be the first in ask about this feature. It could be fantastic, if youd could analyze the possibility to implement it. I base on it several issues designed in CybSim (Physiological - Cybernetic modeling tool). I used a similar feature when I worked with EcosimPro language to develop PhysPK (a different and proprietary Physiological modelling tool), and perhaps this is the reason to feel that it is an interesting feature. I think that Modelica overcomes EcosimPro language in near all features, but there is some feature that I feel missed.

Thanks for your help!!

**Reply by perost** [reply]
It's now implemented in #10732, it should be included in tomorrows nightly build if you want to try it out.

**Comment by casella** [comment]
I opened #10685 on this topic.

**Reply by perost** [reply, marked as answer]
I'd already opened #10681 for that.

**Comment by casella** [comment]
@mpradovelasco, issue #10681 is now closed. You can try with the next nightly build tomorrow.

**Comment by mpradovelasco** [comment]
Hi @casella, I have tested it with success in my project using OMEdit 1.22 dev from yersterday (It has been the first time that I build a OMdev version, following your build OpenModelica instruction from the web). 

Thank very much for your help!
